Tests/CookiesTests.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System.Net.Http.Headers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+using Moq;
+using Cookies = UI.Pages.Cookies;
+
+
+
+namespace Tests;
+
+[Collection(nameof(SliceFixture))]
+public class CookiesTests : CleanSliceFixture
+{
+    public CookiesTests(SliceFixture fixture) : base(fixture)
+    {
+    }
+
+    [Theory]
+    [InlineData(true, true)]
+    [InlineData(null, true)]
+    [InlineData(true, null)]
+    [InlineData(null, null)]
+    public void Cookies_OnGet_Test(bool? consent, bool? preferencesSet)
+    {
+        var httpContext = new DefaultHttpContext();
+        var modelState = new ModelStateDictionary();
+        var actionContext = new ActionContext(httpContext, new RouteData(), new PageActionDescriptor(), modelState);
+        var pageContext = new PageContext(actionContext);
+        Cookies cookiePage = new()
+        {
+            PageContext = pageContext
+        };
+        var result = cookiePage.OnGet(consent, preferencesSet);
+        result.Should().BeOfType<PageResult>();
+    }
+
+    [Fact]
+    public async void Cookies_For_FindATuitionPartner_Created_OnGet()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.OnGet(true, true);
+                  return Task.FromResult(page);
+              });
+        cookiePage.Response.Headers.TryGetValue("Set-Cookie", out var cookieVariables);
+        cookieVariables.ToString().Should().Contain(".FindATuitionPartner.Consent=True");
+    }
+
+    [Fact]
+    public async void Cookies_For_FindATuitionPartner_Retrieved_OnGet()
+    {
+        var cookiePage = await Fixture.GetPage<Cookies>()
+              .Execute(page =>
+              {
+                  page.PageContext = GetPageContext();
+                  page.Request.Cookies = MockRequestCookieCollection(".FindATuitionPartner", "true");
+                  page.OnGet(true, true);
```

Comment:
I think the correct check here is that it uses the cookie value when OnGet(null, <any>) is called. I think you need to check that page.Consent is set to the value from the cookie, not that the cookie is set

Reply:
Changed as asked